### PR TITLE
Split the USB descriptors into a separate file.

### PR DESCRIPTION
**Pull request by nomis52**
This allows us to create board specific hardware configs without duplicating
all the USB data.

### CODE REVIEW

File: firmware/ja-rule.X/nbproject/configurations.xml [hunk 1]
```diff
@@ -55,6 +55,7 @@
         <itemPath>../../common/bootloader_options.h</itemPath>
         <itemPath>../../common/dfu_constants.h</itemPath>
         <itemPath>../../common/reset.h</itemPath>
+        <itemPath>../src/usb_descriptors.h</itemPath>
```
Comment (peternewman) [inline firmware/ja-rule.X/nbproject/configurations.xml]: Is it worth ordering these, or at least grouping src and then common?

File: firmware/ja-rule.X/nbproject/configurations.xml [hunk 2]
```diff
@@ -151,6 +152,7 @@
         <itemPath>../../common/bootloader_options.c</itemPath>
         <itemPath>../../common/uid_store.c</itemPath>
         <itemPath>../../common/reset.c</itemPath>
+        <itemPath>../src/usb_descriptors.c</itemPath>
```
Comment (peternewman) [inline firmware/ja-rule.X/nbproject/configurations.xml]: Again

File: firmware/src/usb_descriptors.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Fetch a pointer to the USB serial number.
+ * @returns a pointer to the unicode string serial number.
+ *
+ * The length of the buffer will be at least UID_LENGTH * 2 + 1, enough to hold
```
Comment (peternewman) [inline firmware/src/usb_descriptors.h]: Is this an @pre or @post?

**Comment by peternewman** [comment]
Generally LGTM

File: firmware/ja-rule.X/nbproject/configurations.xml (hunk 1 above)
Comment (nomis52) [inline firmware/ja-rule.X/nbproject/configurations.xml]: It's sort of up to whatever MPLAB X does. I can try sorting

File: firmware/ja-rule.X/nbproject/configurations.xml (hunk 2 above)
Comment (nomis52) [inline firmware/ja-rule.X/nbproject/configurations.xml]: Done

File: firmware/src/usb_descriptors.h (hunk 3 above)
Comment (nomis52) [inline firmware/src/usb_descriptors.h]: Neither I think.

**Pull request merged**